src/test/regress/expected/multi_subquery_window_functions.out

```diff
@@ -0,0 +1,1049 @@
+ 46.2061855670103093 | Mon Jan 20 00:00:00 2014 |       8
+ 52.5945945945945946 | Mon Jan 20 00:00:00 2014 |       7
+ 52.2589285714285714 | Tue Jan 21 00:00:00 2014 |       6
+(7 rows)
+
+-- aggregates in the PARTITION BY is also allows
+SELECT
+   avg(user_id), max(time), my_rank
+FROM
+(
+  SELECT
+    user_id, date_trunc('day', time) as time, rank() OVER my_win as my_rank
+  FROM
+    events_table
+  GROUP BY
+    user_id, date_trunc('day', time)
+    WINDOW my_win AS (PARTITION BY user_id, avg(event_type%10)::int ORDER BY count(*) DESC)
+) as foo
+WHERE
+  my_rank > 0
+GROUP BY
+  my_rank
+ORDER BY
+  3 DESC, 1 DESC,2 DESC
+LIMIT
+  10;
+         avg         |           max            | my_rank 
+---------------------+--------------------------+---------
+ 22.0000000000000000 | Fri Jan 10 00:00:00 2014 |       8
+ 60.4000000000000000 | Mon Jan 20 00:00:00 2014 |       7
+ 55.5500000000000000 | Tue Jan 21 00:00:00 2014 |       6
+ 50.6142857142857143 | Tue Jan 21 00:00:00 2014 |       5
+ 53.6697247706422018 | Tue Jan 21 00:00:00 2014 |       4
+ 49.7604166666666667 | Tue Jan 21 00:00:00 2014 |       3
+ 47.9569892473118280 | Tue Jan 21 00:00:00 2014 |       2
+ 49.5859375000000000 | Tue Jan 21 00:00:00 2014 |       1
+(8 rows)
+
+-- GROUP BY should not necessarly be inclusive of partitioning
+-- but this query doesn't make much sense
+SELECT
+   avg(user_id), my_rank
+FROM
+(
+  SELECT
+    user_id, rank() OVER my_win as my_rank
+  FROM
+    events_table
+  GROUP BY
+    user_id
+    WINDOW my_win AS (PARTITION BY user_id, max(event_type) ORDER BY count(*) DESC)
+) as foo
+GROUP BY
+  my_rank
+ORDER BY
+  2 DESC, 1 DESC
+LIMIT
+  10;
+         avg         | my_rank 
+---------------------+---------
+ 50.0000000000000000 |       1
+(1 row)
+
+  -- Using previously defined supported window function on distribution key
+SELECT * FROM (
+  SELECT
+    user_id, date_trunc('day', time) as time, sum(rank) OVER w2
+  FROM (
+    SELECT DISTINCT
+      user_id as user_id, time, rank() over w1
+    FROM users_table
+    WINDOW
+      w AS (PARTITION BY user_id),
+      w1 AS (w ORDER BY value_2, value_3)
+  ) fab
+  WINDOW
+    w2 as (PARTITION BY user_id, time)
+) a
+ORDER BY
+  1, 2, 3 DESC
+LIMIT
+  10;
+ user_id |           time           | sum 
+---------+--------------------------+-----
+       0 | Fri Jan 10 00:00:00 2014 |  32
+       0 | Sat Jan 11 00:00:00 2014 |  40
+       0 | Sat Jan 11 00:00:00 2014 |  38
+       0 | Sat Jan 11 00:00:00 2014 |  30
+       0 | Sun Jan 12 00:00:00 2014 |  49
+       0 | Sun Jan 12 00:00:00 2014 |  47
+       0 | Sun Jan 12 00:00:00 2014 |  34
+       0 | Sun Jan 12 00:00:00 2014 |  29
+       0 | Sun Jan 12 00:00:00 2014 |  24
+       0 | Sun Jan 12 00:00:00 2014 |  20
+(10 rows)
+
+-- test with reference table partitioned on columns from both
+SELECT *
+FROM
+(
+    SELECT
+      DISTINCT user_id, it_name, count(id) OVER (PARTITION BY user_id, id)
+    FROM
+      users_table, users_ref_test_table
```

Comment:
This query doesn't make sense at all. Could you add `WHERE users_table.value_2 = users_ref_test_table. k_no ` or such.